packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -0,0 +1,570 @@
+  void pushHandleLayers(LayerLink? startHandle, LayerLink? endHandle);
+
+  /// Gets the selected content in this object.
+  ///
+  /// Return `null` if nothing is selected.
+  SelectedContent? getSelectedContent();
+
+  /// Handles the [SelectionEvent] sent to this object.
+  ///
+  /// The subclasses need to update their selections or delegate the
+  /// [SelectionEvent]s to its subtree.
+  ///
+  /// The `event`s are subclasses of [SelectionEvent]. Use runtime type check to
+  /// determine what kinds of event are dispatched to this handler and handle
+  /// them accordingly.
+  SelectionResult dispatchSelectionEvent(SelectionEvent event);
+}
+
+/// The selected content in a [Selectable].
+class SelectedContent {
+  /// Creates a selected content object.
+  ///
+  /// Only supports plain text.
+  const SelectedContent({required this.plainText});
+
+  /// The selected content in plain text format.
+  final String plainText;
+}
+
+/// A mixin that can be selected by users when under a [SelectionArea] widget.
+///
+/// This object receives selection events and the [value] must reflect the
+/// current selection in this [Selectable]. The object must also notify its
+/// listener if the [value] ever changes.
+///
+/// This object is responsible for drawing selection highlight.
+///
+/// In order to receive the selection event, the mixer need to register
+/// themselves to [SelectionRegistrar]s. Use the
+/// [SelectionRegistrarScope.maybeOf] to get the the selection registrar, and
+/// mix the [SelectionRegistrant] to subscribe to the [SelectionRegistrar]
+/// automatically.
+///
+/// The mixer also need to paints [LayerLink]s of selection handles in a
+/// mobile application.
+///
+/// {@macro flutter.rendering.SelectionHandler}
+///
+/// See also:
+///  * [SelectionArea]: which provides the overview of selection system.
+mixin Selectable implements SelectionHandler {
+  /// {@macro flutter.rendering.RenderObject.getTransformTo}
+  Matrix4 getTransformTo(RenderObject? ancestor);
+
+  /// The size of this [Selectable].
+  Size get size;
+
+  /// The dispose method to enable mixer to use [SelectionRegistrant].
+  void dispose();
+}
+
+/// A mixin to auto-register the mixer to the [registrar].
+///
+/// To use this mixin, the mixer needs to set the [registrar] to the
+/// [SelectionRegistrar] it wants to register to.
+///
+/// This mixin only registers the mixer with the [registrar] if the
+/// [SelectionGeometry.hasContent] returned by the mixer is true.
+mixin SelectionRegistrant on Selectable {
+  /// The [SelectionRegistrar] the mixer will be or is registered to.
+  ///
+  /// This [Selectable] only registers mixer if the
+  /// [SelectionGeometry.hasContent] returned by the [Selectable] is true.
+  SelectionRegistrar? get registrar => _registrar;
+  SelectionRegistrar? _registrar;
+  set registrar(SelectionRegistrar? value) {
+    if (value == _registrar)
+      return;
+    if (value == null) {
+      // when registrar go from non-null to null;
```

Comment:
Capitalize "When" here and below, and maybe periods instead of semicolons if that wasn't on purpose.